smack-core/src/main/java/org/jivesoftware/smack/c2s/ModularXmppClientToServerConnection.java

```diff
@@ -1086,4 +1093,8 @@ public void appendStatsTo(ExtendedAppendable appendable) throws IOException {
         }
 
     }
+
+    public interface StreamOpenFactory {
+        StreamOpen getStreamOpen();
```

Comment:
Factory methods are usually prefixed with "create", e.g. `createFoo()`, not "get". Since they are factories, they create new objects. Hence this should be `createStreamOpen()`.

Furthermore, the method should have parameters, i.e. everything that is required to create a new stream open. Those are basically the same as [this StreamOpen concstructor](https://github.com/igniterealtime/Smack/blob/45e865757a7242ea4942855213b53a9670c06f9c/smack-core/src/main/java/org/jivesoftware/smack/packet/StreamOpen.java#L71) (minus the StreamContentNamespace, which is always 'jabber:client').